### PR TITLE
feat: migrating to pnpm package manager

### PR DESCRIPTION
**Pull request by axel7083**
- [x] Updating the worflows by removing yarn
- [x] Adding `pnpm-lock.yaml`
- [x] Adding `packageManager` to the `package.json` 
- [x] Removing yarn.lock

Fixes https://github.com/containers/podman-desktop-extension-minikube/issues/106

### CODE REVIEW

**Comment by benoitf** [comment]
you may add a reference to https://github.com/containers/podman-desktop-extension-minikube/issues/106

File: .github/workflows/pr-check.yaml [hunk 1]
```diff
@@ -25,66 +25,38 @@ jobs:
...
       matrix:
-        include:
-          - os: "windows-2022"
-          - os: "macos-14"
-          - os: "ubuntu-22.04"
-    timeout-minutes: 20
-    env:
-      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        os: [windows-2022, ubuntu-22.04, macos-14]
+    timeout-minutes: 15
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml:29]: it seems it's not related to yarn/pnpm --> another PR

File: .github/workflows/pr-check.yaml [hunk 2]
```diff
@@ -25,66 +25,38 @@ jobs:
...
+        run: pnpm format:check
 
-      - name: Run Tests
-        run: yarn test
+      - name: Run tests
+        run: pnpm test
 
-      - name: Run Build
-        run: yarn build
-        timeout-minutes: 40
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml:90]: why is it removed

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -97,28 +97,21 @@ jobs:
...
 
-      - uses: actions/cache@v4
-        id: yarn-cache
+      - uses: actions/setup-node@v4
         with:
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
+          node-version: 20
```
Comment (benoitf) [inline .github/workflows/release.yaml:107]: updating to node v20 is another PR

File: package.json [hunk 4]
```diff
@@ -101,11 +101,13 @@
...
+    "globals": "^15.10.0",
     "prettier": "^3.3.3",
     "typescript": "5.6.3",
     "typescript-eslint": "^8.8.1",
     "validator": "^13.12.0",
     "vite": "^5.4.8",
     "vitest": "^2.0.5"
-  }
+  },
+  "packageManager": "pnpm@9.9.0+sha512.60c18acd138bff695d339be6ad13f7e936eea6745660d4cc4a776d5247c540d0edee1a563695c183a66eb917ef88f2b4feb1fc25f32a7adcadc7aaf3438e99c1"
```
Comment (benoitf) [inline package.json:112]: looks like the latest is 9.12.1

File: .github/workflows/pr-check.yaml (hunk 1 above)
Comment (axel7083) [reply .github/workflows/pr-check.yaml]: Restored

File: .github/workflows/pr-check.yaml (hunk 2 above)
Comment (axel7083) [reply .github/workflows/pr-check.yaml:90]: Restored

File: .github/workflows/release.yaml (hunk 3 above)
Comment (axel7083) [reply .github/workflows/release.yaml]: Restored 18

File: package.json (hunk 4 above)
Comment (axel7083) [reply package.json]: Updated

File: package.json [hunk 5]
```diff
@@ -101,11 +101,13 @@
     "eslint-plugin-sonarjs": "^2.0.2",
     "eslint-plugin-unicorn": "^56.0.0",
     "mkdirp": "^3.0.1",
+    "globals": "^15.10.0",
```
Comment (benoitf) [inline package.json:104]: and for globals it's 15.11.0

Comment (axel7083) [reply package.json]: upgraded

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
looks like there are some failures in ci

**Comment by axel7083** [comment]
Sidenote: to update the packageManager, you need to run the command `corepack up` in the root directory, you cannot use the npm registry integrity sha512 value.

**Review by benoitf — APPROVED** [review]

**Pull request merged**